site/src/docs/typechat-in-5-minutes.md

```diff
@@ -0,0 +1,147 @@
+You can call these directly.
+
+```ts
+export function createOpenAILanguageModel(apiKey: string, model: string, endPoint? string): TypeChatLanguageModel;
+
+export function createAzureOpenAILanguageModel(apiKey: string, endPoint: string): TypeChatLanguageModel;
+```
+
+For even more convenience, TypeChat also provides a function to infer whether you're using OpenAI or Azure OpenAI.
+
+```ts
+export function createLanguageModel(env: Record<string, string | undefined>): TypeChatLanguageModel
+```
+
+You can populate your environment variables, and based on whether `OPENAI_API_KEY` or `AZURE_OPENAI_API_KEY` is set, you'll get a model of the appropriate type.
+
+```ts
+import dotenv from "dotenv";
+dotenv.config(/*...*/);
+import * as typechat from "typechat";
+const model = typechat.createLanguageModel(process.env);
+```
+
+Regardless, of how you decide to construct your model, we recommend keeping your secret tokens/API keys in a `.env` file, and specifying `.env` in a `.gitignore`.
+You can use a library like [`dotenv`](https://www.npmjs.com/package/dotenv) to help load these up as we did above.
+
+## Loading the Schema
+
+The first big idea around TypeChat is that we need to provide a set of type definitions to guide AI responses.
+That means that we'll need to literally load up an input `.ts` file verbatim.
+
+```ts
+// Load up the type from our schema.
+import type { SentimentResponse } from "./sentimentSchema";
+
+// Load up the schema file contents.
+const schema = fs.readFileSync(path.join(__dirname, "sentimentSchema.ts"), "utf8");
+```
+
+This introduces some complications to certain kinds of builds, since our input files need to be treated as local assets.
+One way to achieve this is to use a runtime or tool like [`ts-node`](https://www.npmjs.com/package/ts-node) to both import the file for its types, as well as read the file contents.
+Another is to use a utility like [`copyfiles`](https://www.npmjs.com/package/copyfiles) to move specific schema files to the output directory.
+If you're using a bundler, there might be custom way to import a file as a raw string as well.
+Regardless, [our examples](https://github.com/microsoft/TypeChat/tree/main/examples) should work with either of the first two options.
+
+## Creating a JSON Translator
+
+TypeChat provides functions to construct two types of *translators*:
+
+- `createJsonTranslator`: these get well-typed JSON data with based on some types, and
+- `createProgramTranslator`: these get a well-typed representation of a program based on some typed API
+
+The first tends to be simpler, so that's what we'll focus on for the remainder of this example.
+
+```ts
+const translator =
+    typechat.createJsonTranslator<SentimentResponse>(model, schema, "SentimentResponse");
+```
+
+A translator takes a model, the schema file contents, and the name of the type within the schema file.
+That type name effectively acts as the "entry-point" of our schema for the language model.

```

Comment:
It might be worth being explicit that your schema can (obviously) contain multiple types, but you must specify the "entry" type.